language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs

```diff
@@ -1,13 +1,22 @@
+{-# LANGUAGE CPP                  #-}
 {-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE TypeOperators        #-}
 {-# LANGUAGE UndecidableInstances #-}
 
-module Language.PlutusCore.Evaluation.Machine.GenericSemigroup where
+module Language.PlutusCore.Evaluation.Machine.GenericSemigroup (GenericSemigroupMonoid(..)) where
+
+-- TODO when we need to support semigroup >=0.19 only, delete this file
```

Comment:
Just delete this as soon as the stackage snapshot includes the right semigroups. CPP is of the devil (and breaks most tooling).

Reply:
Can't we just extra-dep a newer semigroup?

Reply:
I tried, it follows with an endless string of dependency mismatches. Maybe time to try again? Newest matching LTS would be 15.0, which seems to be a bit new. https://www.stackage.org/package/semigroups/snapshots

Reply:
Yea, LTS-15 isn't supported yet due to the shenanigans stackage pulled with the pantry switch. :-/